crates/bevy_input/src/gamepad.rs

```diff
@@ -11,8 +11,34 @@ pub struct Gamepad(pub usize);
 /// Container of unique connected [Gamepad]s
 ///
 /// [Gamepad]s are registered and deregistered in [gamepad_connection_system]
-pub struct GamepadLobby {
-    pub gamepads: HashSet<Gamepad>,
+pub struct Gamepads {
+    gamepads: HashSet<Gamepad>,
+}
+
+impl Gamepads {
+    /// Returns true if the [Gamepads] contains a [Gamepad].
+    pub fn contains(&self, gamepad: &Gamepad) -> bool {
+        self.gamepads.contains(gamepad)
+    }
+
+    /// Iterates over registered [Gamepad]s
+    pub fn iter(&self) -> std::collections::hash_set::Iter<Gamepad> {
+        self.gamepads.iter()
+    }
+
+    /// Registers [Gamepad].
+    ///
+    /// If this [Gamepad] was present, `true` is returned.
+    ///
+    /// If this [Gamepad] wasn't present, `false` is returned.
+    fn register(&mut self, gamepad: Gamepad) -> bool {
+        self.gamepads.insert(gamepad)
+    }
+
+    /// Deregisters [Gamepad] and returns whether the value was registered
+    fn deregister(&mut self, gamepad: &Gamepad) -> bool {
+        self.gamepads.remove(gamepad)
+    }
```

Comment:
I don't think that these functions need return whether the Gamepad existed,
After all only the `gamepad_connection_system` uses them and does ignore the bools.

```suggestion
    /// Registers [Gamepad].
    fn register(&mut self, gamepad: Gamepad) {
        self.gamepads.insert(gamepad);
    }

    /// Deregisters [Gamepad.
    fn deregister(&mut self, gamepad: &Gamepad)  {
        self.gamepads.remove(gamepad);
    }
```

Reply:
Agreed. The extra return just adds complexity for no tangible gain.